lib/amazon.js

```diff
@@ -66,13 +73,15 @@ function updateRuntime ({ functionName }) {
   return lambda.updateFunctionConfiguration(params).promise().then(res => res.FunctionArn)
 }
 
-exports.updateFunction = async function ({ zipFile, functionName }) {
+exports.updateFunction = async function ({ zipFile, functionName, bucket, s3Key }) {
   verifyZipFile(zipFile)
 
   const params = {
     FunctionName: functionName,
     Publish: true,
-    ZipFile: zipFile
+    ZipFile: !bucket ? zipFile : null,
```

Comment:
Maybe this function can take a `code` object instead of `zipFile`, `bucket` and `s3Key`. The `code` object would be the same as my previous comment, and could just be splated in here:

```js
{
  FunctionName: functionName,
  Publish: true,
  ...code
}
```

Reply:
Yess!!